runtime/compilers/rillv1/parse_connector.go

```diff
@@ -26,5 +28,26 @@ func (p *Parser) parseConnector(node *Node) error {
 
 	r.ConnectorSpec.Driver = tmp.Driver
 	r.ConnectorSpec.Properties = tmp.Defaults
+	r.ConnectorSpec.PropertiesFromVariables, err = propertiesFromVariables(tmp.Defaults)
+	if err != nil {
+		return err
+	}
 	return nil
 }
+
+func propertiesFromVariables(props map[string]string) (map[string]string, error) {
+	res := make(map[string]string)
+	for key, val := range props {
+		meta, err := AnalyzeTemplate(val)
+		if err != nil {
+			return nil, err
+		}
+		// assume that only one variable will be set
+		for _, k := range meta.Variables {
+			if after, found := strings.CutPrefix(k, "vars."); found {
+				res[key] = after
+			}
+		}
+	}
```

Comment:
Can we return an error if there are non-templated contents or multiple variables (without making the code too ugly)? It would be a very hard issue for the user to debug otherwise.